src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObsoletedInPlatformAttribute.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Marks APIs that were obsoleted in a given operating system version.
+    ///
+    /// Primarily used by OS bindings to indicate APIs that should only be used in
+    /// earlier versions.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple = true, Inherited = false)]
+    public sealed class ObsoletedInPlatformAttribute : PlatformAttribute
+    {
+        public ObsoletedInPlatformAttribute(string platformName, string url, string message) : base(platformName)
```

Comment:
The url and message are both required?  Wondering if either or both should be nullable.  Same goes for the corresponding properties.

Reply:
The API shape should be:

```C#
    public sealed class ObsoletedInOSPlatformAttribute : OSPlatformAttribute
    {
        public ObsoletedInPlatformAttribute(string platformName);
        public ObsoletedInPlatformAttribute(string platformName, string message);
        public string Message { get; }
        public string Url { get; set; }
    }
```

https://github.com/dotnet/runtime/issues/33331#issuecomment-650326500